docs/src/references/mqtt-api.md

```diff
@@ -0,0 +1,838 @@
+
+```sh te2mqtt
+tedge mqtt sub 'te/device/main/+/+/m/+'
+```
+
+Or to subscribe to a specific type of measurement published to an services on the main device, then use:
+
+```sh te2mqtt
+tedge mqtt sub 'te/device/main/service/+/m/memory'
+```
+
+#### Publish to a service on a child device
+
+```sh te2mqtt
+tedge mqtt pub -r 'te/device/child01/service/nodered/m/environment' '{
+  "temperature": 23.4
+}'
+```
+
+### Telemetry type metadata
+
+The data types also may have additional metadata associated with it,
+which can be added/updated by publishing to `/meta` subtopics of those data types.
+For example, the units associated with measurements in the `battery_reading` measurement type
+can be updated by publishing the following message:
+
+```sh te2mqtt
+tedge mqtt pub -r te/device/main///m/battery_reading/meta '{
+  "units": {
+    "temperature": "°C",
+    "voltage": "V",
+    "current": "A"
+  }
+}'
+```
+
+The metadata fields supported by each data type will be defined in detail later.
+
+## Commands
+
+The topic scheme for commands can be visualized using the diagram below. The **identifier** is used as the command 
+
+<p align="center">
+
+```mermaid
+graph LR
+  te --/--- identifier --/--- cmd
+  subgraph root
+    te
+  end
+
+  subgraph identifier
+    identifier2["&lt;identifier&gt;"]
+  end
+
+  subgraph command
+    direction LR
+    cmd --/--- cmd_type["&lt;cmd_type&gt;"] --/--- cmd_id["&lt;cmd_id&gt;"]
+  end
+
+```
+
+</p>
+
+Where the command segments are describe as follows:
+
+|Segment|Description|
+|---|----|
+|&lt;identifier&gt;|The [identifier](#group-identifier) (e.g. device/service) associated with the command.|
+|&lt;cmd_type&gt;|Command type. Each command can define its own payload schema to allow commands to have parameters related to the command's function.|
+|&lt;cmd_id&gt;|Unique command id which is unique for the command instance. e.g. `123456`, `d511a86cab95be81` etc.|
+
+### Command examples
+
+The following table details some example command types which are supported by thin-edge.
+
+|Command Type|Example Topic|
+|------|---|
+|software_list|`te/<identifier>/cmd/software_list/<cmd_id>`|
+|software_update|`te/<identifier>/cmd/software_update/<cmd_id>`|
+|config_snapshot|`te/<identifier>/cmd/config_snapshot/<cmd_id>`|
+|config_update|`te/<identifier>/cmd/config_update/<cmd_id>`|
+|firmware_update|`te/<identifier>/cmd/firmware_update/<cmd_id>`|
+|restart|`te/<identifier>/cmd/restart/<cmd_id>`|
+
+The command would be interpreted differently based on the target entity.
+For example, the `restart` could mean either a device restart or a service restart based on the target entity.
+
+### Examples: With default device/service topic semantics
+
+#### Command to main device
+
+Command to fetch the software list from the main device:
+
+```sh te2mqtt
+tedge mqtt pub -r 'te/device/main///cmd/software_list/123' '{
+  "status": "init"
```

Comment:
@didier-wenzek would it be ok if we allow the `.status` to have a default value of `"init"` if it is not set in the first command creation?

For example, the first message to start the `software_list` command could be:

```sh
tedge mqtt pub -r 'te/device/main///cmd/software_list/123' '{}'
```

This reduces the amount of information that the user/component needs to know when creating an command message.

Reply:
Or even empty ``, not just empty JSON: `{}`.